metricflow/test/snapshots/test_dataflow_to_sql_plan.py/SqlQueryPlan/DuckDB/test_derived_metric_with_offset_to_grain__plan0_optimized.sql

```diff
@@ -29,31 +29,26 @@ FROM (
   ) subq_18
   INNER JOIN (
     -- Join to Time Spine Dataset
+    -- Pass Only Elements:
+    --   ['bookings', 'metric_time__day']
+    -- Aggregate Measures
+    -- Compute Metrics via Expressions
     SELECT
-      subq_25.ds AS metric_time__day
-      , subq_23.bookings_at_start_of_month AS bookings_at_start_of_month
-    FROM ***************************.mf_time_spine subq_25
+      subq_22.ds AS metric_time__day
+      , SUM(subq_20.bookings) AS bookings_at_start_of_month
+    FROM ***************************.mf_time_spine subq_22
     INNER JOIN (
-      -- Aggregate Measures
-      -- Compute Metrics via Expressions
+      -- Read Elements From Semantic Model 'bookings_source'
+      -- Metric Time Dimension 'ds'
       SELECT
-        metric_time__day
-        , SUM(bookings) AS bookings_at_start_of_month
```

Comment:
Ok, this is where we're hitting the problem. Before, we would aggregate this metric per day and then do the join. Now, we do the join and then compute the metric. But in doing that, we explode the set of input rows to the metric.

I know the description lays out a reason for switching the join order, but what the original code was doing - which the updated code does not handle - was pre-aggregating in order to avoid inappropriate fanout on the time spine join. Without that pre-aggregation you cannot get correct results for expanded granularities for offset_to_grain on additive measures, because you end up over-counting values.